service_management/azure/test/integration/service_bus/queues_test.rb

```diff
@@ -14,154 +14,196 @@
+      queue = subject.create_queue name_alternative, description_alternative
+
+      queue.must_be_kind_of Azure::ServiceBus::Queue
+      queue.name.must_equal name_alternative
+      queue.lock_duration.must_equal 30.0
+      queue.max_size_in_megabytes.must_equal description_alternative[:max_size_in_megabytes]
+      queue.requires_duplicate_detection.must_equal description_alternative[:requires_duplicate_detection]
+      queue.requires_session.must_equal description_alternative[:requires_session]
+      queue.default_message_time_to_live.must_equal 1800.0
+      queue.dead_lettering_on_message_expiration.must_equal description_alternative[:dead_lettering_on_message_expiration]
+      queue.duplicate_detection_history_time_window.must_equal 1200.0
+      queue.max_delivery_count.must_equal description_alternative[:max_delivery_count]
+      queue.enable_batched_operations.must_equal description_alternative[:enable_batched_operations]
+    end
+
+    it "should create a new queue from a Queue with a description hash" do
+      queue = subject.create_queue Azure::ServiceBus::Queue.new(name_alternative, description_alternative)
+
+      queue.must_be_kind_of Azure::ServiceBus::Queue
+      queue.name.must_equal name_alternative
+      queue.lock_duration.must_equal 30.0
+      queue.max_size_in_megabytes.must_equal description_alternative[:max_size_in_megabytes]
+      queue.requires_duplicate_detection.must_equal description_alternative[:requires_duplicate_detection]
+      queue.requires_session.must_equal description_alternative[:requires_session]
+      queue.default_message_time_to_live.must_equal 1800.0
+      queue.dead_lettering_on_message_expiration.must_equal description_alternative[:dead_lettering_on_message_expiration]
+      queue.duplicate_detection_history_time_window.must_equal 1200.0
+      queue.max_delivery_count.must_equal description_alternative[:max_delivery_count]
+      queue.enable_batched_operations.must_equal description_alternative[:enable_batched_operations]
+    end
   end
 
   describe 'when a queue exists' do
-    before { subject.create_queue queue_name }
+    before do
+      VCR.insert_cassette "service_bus/#{name}"
+      subject.create_queue queue_name
+    end
 
-    describe '#delete_queue' do
-      it "should raise exception if the queue cannot be deleted" do
-        assert_raises(Azure::Core::Http::HTTPError) do
-          subject.delete_queue ServiceBusQueueNameHelper.name
-        end
-      end
+    after do
+      subject.delete_queue queue_name
+      VCR.eject_cassette
+    end
 
-      it "should be able to delete the queue" do
-        response = subject.delete_queue queue_name
-        response.must_equal nil
-      end
+    it "should get a queue by name" do
+      result = subject.get_queue queue_name
+
+      result.must_be_kind_of Azure::ServiceBus::Queue
+      result.name.must_equal queue_name
+    end
+
+    it "should be able to delete the queue" do
```

Comment:
is this test different from the create/delete that happens in the before/after block?